hbase-zookeeper/src/main/java/org/apache/hadoop/hbase/zookeeper/RecoverableZooKeeper.java

```diff
@@ -726,6 +726,13 @@ public synchronized States getState() {
   }
 
   public synchronized ZooKeeper getZooKeeper() {
+    // Callers expect an initialized ZooKeeper instance
+    // Pre HBASE-28529 the constructor used to call checkZk()
+    try {
+      checkZk();
+    } catch (Exception x) {
+      // Just return null zk
```

Comment:
Better log the exception and return null?

Reply:
OK